src/Features/LanguageServer/Protocol/Handler/SemanticTokens/SemanticTokensHelpers.cs

```diff
@@ -272,7 +272,12 @@ private static ClassifiedSpan[] ConvertMultiLineToSingleLineSpans(SourceText tex
                     }
 
                     var updatedClassifiedSpan = new ClassifiedSpan(textSpan.Value, classificationType);
-                    updatedClassifiedSpans.Add(updatedClassifiedSpan);
+
+                    // Omit 0-length spans created in this fashion.
+                    if (updatedClassifiedSpan.TextSpan.Length != 0)
```

Comment:
If the updatedClassifiedSpan.TextSpan == textSpan in practice (I think it does) you could prevent creating the updatedClassifiedSpan entirely and do a `if (textSpan.Value.Length != 0) ....`